### ISSUE TITLE
Feature Request: fail when a rule at level "warn" doesn't trigger.

### ISSUE THREAD

**Issue by imoverclocked** [google/error-prone#2664]
In a large installation with sub-projects, sometimes some projects break different rules than others. If the goal is to eliminate all known issues in a codebase, some sub-projects may set a rule to `warn` while the issues are worked out.

```
allProjects:
  error("ReturnsNullCollection")

legacyProject:
  warn("ReturnsNullCollection")
```

It would be nice for maintainers to be able to validate that all rules specified at `warn` are actually still needed at that level.

```
legacyProject:
  // must be removed when last instance of this rule is fixed in legacyProject.
  mustWarn("ReturnsNullCollection")
```

**Comment by tbroyer** [comment]
How would that play out with incremental compilation? https://docs.gradle.org/current/userguide/java_plugin.html#sec:incremental_compile

**Comment by imoverclocked** [comment]
Good question! From the [userguide](https://docs.gradle.org/current/userguide/java_plugin.html#sec:incremental_compile):

> A class’s dependencies are determined from type references in its bytecode or symbol analysis via a compiler plugin.

I assume that changing this setting would invalidate the cache. Potentially, when the setting is enabled, errorprone could invalidate on every compile. While this is slow, it can be marked as a maintainer option.

(the setting might look different from above, more like `-XepConfiguredWarningsMustWarn`)
